jetstream_pt/attention_kernel.py

```diff
@@ -0,0 +1,261 @@
+import jax
+import jax.numpy as jnp
+from jax.experimental import pallas as pl
+from jax.experimental.pallas import tpu as pltpu
+from jax.experimental.shard_map import shard_map
+
+import numpy as np
+
+DEFAULT_MASK_VALUE = -0.7 * float(np.finfo(np.dtype("float32")).max)
+
+def ragged_flash_attention_kernel(
+    start_ref,
+    end_ref,
+    line_end_ref,
+    pre_b_ref,
+    pre_i_ref,
```

Comment:
q, k, v and related are easy to read. What are the b, i, o, m, l, bk and pre? Can you add brief description to describe them?

Reply:
Added. Done.